packages/core/src/connectors/facebook/index.ts

```diff
@@ -0,0 +1,135 @@
+import { z } from 'zod';
+
+import {
+  accessTokenEndpoint,
+  authorizationEndpoint,
+  scope,
+  userInfoEndpoint,
+} from '@/connectors/facebook/constant';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '@/connectors/types';
+import { getConnectorConfig, getConnectorRequestTimeout } from '@/connectors/utilities';
+
+export const metadata: ConnectorMetadata = {
+  id: 'facebook',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Facebook',
+    zh_CN: 'Facebook 登录',
+  },
+};
+
+const facebookConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type FacebookConfig = z.infer<typeof facebookConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  if (!config) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, 'Missing config');
+  }
+
+  const result = facebookConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const config = await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    client_id: config.clientId,
+    redirect_uri: redirectUri,
+    response_type: 'code',
+    state,
+    scope, // Only support fixed scope for v1.
+  })}`;
+};
+
+export const getAccessToken: GetAccessToken = async (code, redirectUri) => {
+  type FacebookAccessTokenResponse = {
+    access_token: string;
+    token_type: string;
+    expires_in: number;
+  };
+
+  const { clientId: client_id, clientSecret: client_secret } =
+    await getConnectorConfig<FacebookConfig>(metadata.id);
+
+  const { access_token: accessToken } = await got
+    .post({
+      url: accessTokenEndpoint,
+      form: {
+        code,
+        client_id,
+        client_secret,
+        redirect_uri: redirectUri,
+      },
+      timeout: await getConnectorRequestTimeout(),
+    })
+    .json<FacebookAccessTokenResponse>();
+
+  if (!accessToken) {
+    throw new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid);
+  }
+
+  return accessToken;
+};
+
+export const getUserInfo: GetUserInfo = async (accessToken: string) => {
+  type FacebookUserInfoResponse = {
+    id: string;
+    email?: string;
+    name?: string;
+    picture?: { data: { url: string } };
+  };
+
+  try {
+    const { id, email, name, picture } = await got
+      .get(userInfoEndpoint, {
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        searchParams: {
+          fields: 'id,name,email,picture',
+        },
+        timeout: await getConnectorRequestTimeout(),
```

Comment:
Might miss the context here why do we need to make this setting as an async func? @xiaoyijun

Reply:
see: https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1646202120785129

Reply:
So keep it?

Reply:
Leave it.